app/views/docs/offline.phtml

```diff
@@ -4,17 +4,57 @@
 </p>
+
+<h2>Enable Offline Support</h2>
+
 <p>
-    You should consider enabling Offline Support if you expect the device running your application to experience short periods without a network connection, like going on an airplane or through a tunnel. Offline Support can't help you if the app's data needs to remain up to date with frequent updates or if write requests need to be processed immediately. If you expect your app to be offline for days or weeks at a time, you may also need to find alternative solutions.
+    Offline Support can be enabled by setting <code>setOffilnePersistency</code> to true on your SDK's client.
+</p>
+
+<div class="notice">
+  <h2>Adding Offline Support to Existing Projects</h2>
+  <p>Enabling Offline Support introduces breaking changes to the behavior of Client SDKs. Asynchronous operations may not resolve until the device is online again. Existing projects will have to be updated.</p>
+  <p><a href="#async-operation">Learn more about asynchronous operations</a></p>
+</div>
+
+[TODO: Code Example]
+<p>
+    Offline Support will cache all read request in a <a href="https://en.wikipedia.org/wiki/Cache_replacement_policies#LRU" target="_blank" rel="noopener">Least Recently Used (LRU)</a> cache, so the information can remain available even with no internet connection. The LRU cache defaults to a size of 16MB and can be configured using the <code>setOfflineCacheSize</code> method.
 </p>
 
-<h3>Considerations and Tradeoffs</h3>
+[TODO: Code Example]
+
+<h2 href="#async-operation" id="async-operation">Asynchronous Operation</h2>
 <p>
-    Offline Support adds a cache layer that takes up additional resources and adds overhead to your requests. This may not be desirable for some applications, especially if it's not expected to be used without a stable internet connection.
+    When Offline Support is enabled, asynchronous write operations only resolve after the server accepts the request. Requests made with the Client SDK may block code execution until network connection is available again.
 </p>
-<h3>Offline Support for Existing Projects</h3>
+[TODO: Show UML diagram]
+
+<p>
+    For example, the code example below will <b>block code execution</b> when the device is offline.
+</p>
+
+[TODO: Show Example, in comments clearly mark that it's ]
+
 <p>
-    Asynchronous operations may not resolve immediately when Offline Support is enabled. Existing projects will require some code to be rewritten when adopting Offline Support. <a href="TBD">Learn more about how to handle asynchronous operation</a>.  
-</p>
\ No newline at end of file
+    To avoid blocking code execution, asynchronous operations should be handled [TODO: fill description]
+</p>
+
+[TODO: Show correct version of example. ]
```

Comment:
To avoid blocking, don't `await`

```dart
    final document = databases.createDocument(
      databaseId: '[DATABASE_ID]',
      collectionId: '[COLLECTION_ID]',
      documentId: ID.unique(),
      data: {},
    );
```

Reply:
@stnguyen90 I was really hoping to show an illustrative example where we also show updating local state/ui optimistically to make the point clear.

Is this something we can do?

Reply:
Updated with more info about state:

```dart
  void nonBlockingSubmitTodo() async {
    if (inputController.text.isEmpty || isLoading) return;
    final messenger = ScaffoldMessenger.of(context);
    final newTodo = Todo(
      content: inputController.text,
      id: ID.unique(),
    );

    _databases.createDocument(
      databaseId: '[DATABASE_ID]',
      collectionId: '[COLLECTION_ID]',
      documentId: newTodo.id,
      data: {
        "content": newTodo.content,
        "isComplete": newTodo.isComplete,
      },
    ).catchError((e) {
      messenger.showSnackBar(createErrorSnackBar(e.toString()));
    });

    setState(() {
      inputController.text = '';
      todos.add(newTodo);
      isLoading = false;
    });
  }
```